module_utils/snapshot_lsr/validate.py

```diff
@@ -0,0 +1,340 @@
+
+    if cmd == SnapshotCommand.SNAPSHOT:
+        rc, message = validate_snapshot_args(module_args)
+    #
+    # Currently check, remove, revert, extend and list don't need extra validation
+    #
+    elif cmd == SnapshotCommand.MOUNT:
+        rc, message = validate_mount_args(module_args)
+    elif cmd == SnapshotCommand.UMOUNT:
+        rc, message = validate_umount_args(module_args)
+
+    if rc != SnapshotStatus.SNAPSHOT_OK:
+        return {"return_code": rc, "errors": message, "changed": False}, None
+
+    rc, message, snapset_dict = get_json_from_args(module, module_args, vg_include)
+    logger.info("validate_snapset_args: END snapset_dict is %s", str(snapset_dict))
+    return {"return_code": rc, "errors": message, "changed": False}, snapset_dict
+
+
+def validate_json_request(snapset_json, check_percent_space_required):
+
+    if "name" not in snapset_json:
+        return (SnapshotStatus.ERROR_JSON_PARSER_ERROR, "snapset must include a name")
+
+    if "volumes" not in snapset_json:
+        return (
+            SnapshotStatus.ERROR_JSON_PARSER_ERROR,
+            "snapset must include a volumes list",
+        )
+
+    for list_item in snapset_json["volumes"]:
+        if "vg" not in list_item:
+
+            return (
+                SnapshotStatus.ERROR_JSON_PARSER_ERROR,
+                "snapset vg entry not found",
+            )
+        if "lv" not in list_item:
+            return (
+                SnapshotStatus.ERROR_JSON_PARSER_ERROR,
+                "snapset lv entry not found",
+            )
+
+        if check_percent_space_required:
+
+            if not list_item.get("percent_space_required"):
+                return (
+                    SnapshotStatus.ERROR_JSON_PARSER_ERROR,
+                    "snapset percent_space_required entry not found",
+                )
+            rc, message = check_required_space(list_item["percent_space_required"])
+            if rc != SnapshotStatus.SNAPSHOT_OK:
+                return rc, message
+
+    return SnapshotStatus.SNAPSHOT_OK, ""
+
+
+def validate_json_mount(snapset_dict):
+    volume_list = snapset_dict["volumes"]
+    for list_item in volume_list:
+
+        if "blockdev" not in list_item and (
+            "vg" not in list_item or "lv" not in list_item
+        ):
+            return (
+                SnapshotStatus.ERROR_MOUNT_INVALID_PARAMS,
+                "must provide vg/lv or blockdev for mount source",
+            )
+
+        if "mountpoint" not in list_item:
+            return (
+                SnapshotStatus.ERROR_MOUNT_INVALID_PARAMS,
+                "must provide mountpoint",
+            )
+
+    return SnapshotStatus.SNAPSHOT_OK, ""
+
+
+def validate_json_umount(snapset_dict):
+    volume_list = snapset_dict["volumes"]
+    for list_item in volume_list:
+        try:
+            vg = list_item["vg"]
+            lv = list_item["lv"]
+        except KeyError:
+            vg = None
```

Comment:
```suggestion
            vg = None
            lv = None
```
Otherwise, lv will either be undefined - NameError - or will use a previous value